Controller/Order/Save.php

```diff
@@ -138,7 +138,7 @@ public function execute()
      * @param Quote $quote
      * @return void
      */
-    private function replaceQuote($quote)
+    protected function replaceQuote($quote)
```

Comment:
Although this is a handy way to solve the problem (and a sort of accepted practice in M2 to use protected methods for easier testing) it is not a good practice in general OOP. You could probably leave the methods as private and set the expectations on `$this->checkoutSession` object methods. Anyway, it is a matter of preference, so I am not requesting the change.

Reply:
Reviewing this, it makes a lot more sense this way. @matt-thomason and I were trying to get this to work before as the `expected` calls refused to work properly and after trial and error we came to this solution. However, without the expectations set on the methods called inside the `private`/`protected` methods the contents aren't actually being tested. Functionally for the top level method being tested (`execute`), it doesn't really matter which private method these `checkoutSession` methods are called it, just that they are called at all.

So with that all said, I'm switching in line with @danac-gs's suggestions because it does actually fix the testing coverage.